src/Tribe/Widget/Widget_Abstract.php

```diff
@@ -0,0 +1,313 @@
+<?php
+
+namespace Tribe\Widget;
+
+use Tribe__Utils__Array as Arr;
+
+/**
+ * The abstract base without Views that all widgets should implement.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Widget
+ */
+abstract class Widget_Abstract extends \WP_Widget implements Widget_Interface {
+
+	/**
+	 * Slug of the current widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $slug;
+
+	/**
+	 * Default arguments to be merged into final arguments of the widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string,mixed>
+	 */
+	protected $default_arguments = [];
+
+	/**
+	 * Array map allowing aliased widget arguments.
+	 *
+	 * The array keys are aliases of the array values (i.e. the "real" widget attributes to parse).
+	 * Example array: [ 'alias' => 'canonical', 'from' => 'to', 'that' => 'becomes_this' ]
+	 * Example widget usage: [some_tag alias=17 to='Fred'] will be parsed as [some_tag canonical=17 to='Fred']
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string,string>
+	 */
+	protected $aliased_arguments = [];
+
+	/**
+	 * Array of callbacks for validation of arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string,callable>
+	 */
+	protected $validate_arguments_map = [];
+
+	/**
+	 * Arguments of the current widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string,mixed>
+	 */
+	protected $arguments;
+
+	/**
+	 * HTML content of the current widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $content;
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function __construct( $id_base = '', $name = '', $widget_options = [], $control_options = [] ) {
+		$arguments = $this->get_arguments();
+
+		parent::__construct(
+			Arr::get( $arguments, 'id_base', '' ),
+			Arr::get( $arguments, 'name', '' ),
+			Arr::get( $arguments, 'widget_options', [] ),
+			Arr::get( $arguments, 'control_options', [] )
+		);
+
+		// @todo add what this does in in TEC-3612 & TEC-3613.
+		$this->setup();
+	}
+
+	/**
+	 * Setup the widget.
+	 *
+	 * @todo update in TEC-3612 & TEC-3613
+	 *
+	 * @since TBD
+	 *
+	 * @return mixed
+	 */
+	public abstract function setup();
+
+	/**
+	 * Echoes the widget content.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $args     Display arguments including 'before_title', 'after_title',
+	 *                                      'before_widget', and 'after_widget'.
+	 * @param array<string,mixed> $instance The settings for the particular instance of the widget.
+	 */
+	public function widget( $args, $instance ) {
+		echo $this->get_html();
+	}
+
+	/**
+	 * Returns the rendered View HTML code.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_html() {
+		return $this->get_view()->get_html();
```

Comment:
Does the view not need to be passed the current args?

Reply:
I think the goal is to have One instance per individual widget, so it pulls from `$this->get_arguments()`